daemon/container_operations_windows.go

```diff
@@ -35,6 +43,62 @@ func detachMounted(path string) error {
 	return nil
 }
 
+func (daemon *Daemon) setupSecretDir(c *container.Container) (setupErr error) {
+	if len(c.SecretReferences) == 0 {
+		return nil
+	}
+
+	localMountPath := c.SecretMountPath()
+	logrus.Debugf("secrets: setting up secret dir: %s", localMountPath)
+
+	defer func() {
+		if setupErr != nil {
+			if err := os.RemoveAll(localMountPath); err != nil {
+				log.Errorf("error cleaning up secret mount: %s", err)
+			}
+		}
+	}()
+
+	// create secrets root
+	system.MkdirAll(localMountPath, 0)
+
+	for _, s := range c.SecretReferences {
+		if c.SecretStore == nil {
+			return fmt.Errorf("secret store is not initialized")
```

Comment:
This is copied from `container_operations_unix.go` and I prefer to keep the files in sync initially, unless there is an error. I can work on making the changes to both files. Maybe the check is inside the loop because, perhaps, it only applies if there is at least one secret reference. I will look into this, as it looks like it should only be checked once.